hail/src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -171,8 +172,11 @@ object BlockMatrix {
       val sb = new StringBuilder(blockSize << 2)
       val paddedIndex = StringUtils.leftPad(index.toString, dRect, "0")
       val outputFile = output + "/rect-" + paddedIndex + "_" + r.mkString("-")
-
-      val osw = new OutputStreamWriter(sHadoopBc.value.value.unsafeWriter(outputFile))
+      val uos = sHadoopBc.value.value.unsafeWriter(outputFile)
+      val os = if (writeBytes)
+          new DoubleOutputBuffer(uos, RichArray.defaultBufSize)
+        else
+          new OutputStreamWriter(uos)
```

Comment:
This seems asymmetric. The OutputStreamWriter buffers, but only as necessary to encode the characters. DoubleOutputBuffer seems to actually buffer? (It's also confusingly in a file called `DoubleInputBuffer`).

I think we should use [`java.io.DataOutputStream`](https://docs.oracle.com/javase/7/docs/api/java/io/DataOutputStream.html) rather than roll our own code. 

It is not clear if `FSDataOutputStream` (the output stream created by hadoop) does buffering already. If this is performance critical code, we should measure the effect of wrapping `uos` in a `BufferedOutputStream`. If there is no buffer between this code and the file, a `BufferedOutputStream` should significantly improve write speed for the text path.